### PR TITLE
Add island shop

### PR DESCRIPTION
**Pull request by dlsf**

### CODE REVIEW

File: plugin/src/main/java/com/iridium/iridiumskyblock/commands/ShopCommand.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            String[] commandArguments = Arrays.copyOfRange(arguments, 1, arguments.length);
+            String categoryName = String.join(" ", commandArguments);
+            Optional<ShopCategory> category = IridiumSkyblock.getInstance().getShopManager().getCategoryByName(categoryName);
+
+            if (!category.isPresent()) {
+                player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().noShopCategory.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+                return;
+            }
+
+            player.openInventory(new ShopCategoryGUI(category.get()).getInventory());
```
Comment (dlsf) [inline plugin/src/main/java/com/iridium/iridiumskyblock/commands/ShopCommand.java:52]: Execute a command here

File: plugin/src/main/java/com/iridium/iridiumskyblock/gui/ShopOverviewGUI.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        }
+
+        String itemName = currentItem.getItemMeta().getDisplayName();
+        Optional<ShopCategory> shopCategory = IridiumSkyblock.getInstance().getShopManager().getCategoryByFormattedName(itemName);
+
+        if (!shopCategory.isPresent()) {
+            return;
+        }
+
+        event.getWhoClicked().openInventory(new ShopCategoryGUI(shopCategory.get()).getInventory());
```
Comment (dlsf) [inline plugin/src/main/java/com/iridium/iridiumskyblock/gui/ShopOverviewGUI.java:39]: Execute command here

File: plugin/src/main/java/com/iridium/iridiumskyblock/commands/ShopCommand.java (hunk 1 above)
Comment (dlsf) [reply plugin/src/main/java/com/iridium/iridiumskyblock/commands/ShopCommand.java:52]: No

File: plugin/src/main/java/com/iridium/iridiumskyblock/commands/ShopCommand.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+     *
+     * @param sender The CommandSender which executes this command
+     * @param arguments The arguments used with this command. They contain the sub-command
+     */
+    @Override
+    public void execute(CommandSender sender, String[] arguments) {
+        Player player = (Player) sender;
+        User user = IridiumSkyblockAPI.getInstance().getUser(player);
+        if (!user.getIsland().isPresent()) {
+            player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().dontHaveIsland.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
```
Comment (PeachesMLG) [inline plugin/src/main/java/com/iridium/iridiumskyblock/commands/ShopCommand.java:43]: Do we need an island to open island shop? shouldnt impact us buying/selling stuff

File: plugin/src/main/java/com/iridium/iridiumskyblock/commands/ShopCommand.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            return;
+        }
+        
+        if (arguments.length == 1) {
+            player.openInventory(new ShopOverviewGUI().getInventory());
+        } else {
+            String[] commandArguments = Arrays.copyOfRange(arguments, 1, arguments.length);
+            String categoryName = String.join(" ", commandArguments);
```
Comment (PeachesMLG) [inline plugin/src/main/java/com/iridium/iridiumskyblock/commands/ShopCommand.java:51]: Is the cagegory name ever going to be more than 1 word? Guess it doesnt really matter but still

**Review by PeachesMLG — COMMENTED** [review]
A few things Id like to change, but ima merge it and change them myself

**Pull request merged**